File: PTA 1001 害死人不偿命的(3n+1)猜想.cpp
/*  PTA 1001 害死人不偿命的(3n+1)猜想
    2022/07/17
*/
#include<iostream>
using namespace std;

int main()
{
	int N, i = 0;
	cin >> N;
	for (; N != 1; i++)
	{
		if (N % 2 == 0)
			N = N / 2;
		else
			N = (3 * N + 1) / 2;
	}
	cout << i;
	return 0;
}

File: PTA 1001 害死人不偿命的(3n+1)猜想.java
/// PTA 1001 害死人不偿命的(3n+1)猜想
/// 2023/01/05

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner myscan = new Scanner(System.in);
        int n = myscan.nextInt();
        int count = 0;
        while (n != 1) {
            if (n % 2 == 0)
                n /= 2;
            else
                n = (3 * n + 1) / 2;
            count++;
        }
        System.out.println(count);
    }
}
